From: Christian Marillat <marillat@debian.org>
Date: Tue, 15 May 2012 04:49:36 -0300
Subject: Enable compilation on Debian arches other than Linux (kFreeBSD,
 HURD).
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

One of the main points here is that udev is present only on Linux (that is,
it is the exception, not the norm). So, we check if LINUX is defined or not
in the appropriate places, instead of listing every operating system that is
not Windows or MacOS X.

Forwarded: yes
Last-Update: 2012-06-07

Reviewed-by: Rogério Theodoro de Brito <rbrito@ime.usp.br>
Reviewed-by: Pino Toscano <pino@debian.org>
Signed-off-by: Rogério Theodoro de Brito <rbrito@ime.usp.br>
---
 gtk/configure.ac       |    9 +++++++--
 gtk/src/callbacks.c    |   11 ++++++++---
 libhb/ports.h          |    2 +-
 make/include/main.defs |    5 +++++
 4 files changed, 21 insertions(+), 6 deletions(-)

--- a/gtk/configure.ac
+++ b/gtk/configure.ac
@@ -94,10 +94,15 @@
 	mingw_flag=yes
     ;;
   *)
+    case "$host_os" in
+	linux*)
+	  pkg_gudev="gudev-1.0"
+	  ;;
+    esac
     if test "x$gst_disable" = "xyes" ; then
-	    GHB_PACKAGES="gtk+-2.0 >= 2.10 gthread-2.0 gio-2.0 libnotify dbus-glib-1 gudev-1.0"
+	    GHB_PACKAGES="gtk+-2.0 >= 2.10 gthread-2.0 gio-2.0 libnotify dbus-glib-1 $pkg_gudev"
     else
-	    GHB_PACKAGES="gtk+-2.0 >= 2.10 gthread-2.0 gstreamer-0.10 gstreamer-interfaces-0.10 gstreamer-video-0.10 gstreamer-pbutils-0.10 gio-2.0 libnotify dbus-glib-1 gudev-1.0"
+	    GHB_PACKAGES="gtk+-2.0 >= 2.10 gthread-2.0 gstreamer-0.10 gstreamer-interfaces-0.10 gstreamer-video-0.10 gstreamer-pbutils-0.10 gio-2.0 libnotify dbus-glib-1 $pkg_gudev"
     fi
 	mingw_flag=no
     ;;
--- a/gtk/src/callbacks.c
+++ b/gtk/src/callbacks.c
@@ -28,7 +28,9 @@
 #if !defined(_WIN32)
 #include <poll.h>
 #define G_UDEV_API_IS_SUBJECT_TO_CHANGE 1
+#if defined(__linux__)
 #include <gudev/gudev.h>
+#endif
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
@@ -49,6 +51,9 @@
 #endif
 
 #include <gdk/gdkx.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #else
 #define WINVER 0x0500
 #include <winsock2.h>
@@ -3943,14 +3948,14 @@
 	return dvd_devices;
 }
 
-#if !defined(_WIN32)
+#if defined(__linux__)
 static GUdevClient *udev_ctx = NULL;
 #endif
 
 gboolean
 ghb_is_cd(GDrive *gd)
 {
-#if !defined(_WIN32)
+#if defined(__linux__)
 	gchar *device;
 	GUdevDevice *udd;
 
@@ -3984,7 +3989,7 @@
 void
 ghb_udev_init()
 {
-#if !defined(_WIN32)
+#if defined(__linux__)
 	udev_ctx = g_udev_client_new(NULL);
 #endif
 }
--- a/libhb/ports.h
+++ b/libhb/ports.h
@@ -51,7 +51,7 @@
 #elif defined( SYS_DARWIN )
 #  define HB_LOW_PRIORITY    0
 #  define HB_NORMAL_PRIORITY 31
-#elif defined( SYS_LINUX ) || defined( SYS_FREEBSD ) || defined ( SYS_SunOS )
+#elif defined( SYS_LINUX ) || defined( SYS_FREEBSD ) || defined ( SYS_SunOS ) || defined ( __FreeBSD_kernel__ )
 #  define HB_LOW_PRIORITY    0
 #  define HB_NORMAL_PRIORITY 0
 #elif defined( SYS_CYGWIN )
--- a/make/include/main.defs
+++ b/make/include/main.defs
@@ -64,6 +64,11 @@
     MODULES += gtk
 endif
 
+ifeq (1-gnu,$(FEATURE.gtk)-$(BUILD.system))
+    ## build gtk when gtk+gnu
+    MODULES += gtk
+endif
+
 ifeq (1,$(FEATURE.local_yasm))
     MODULES += contrib/yasm
 endif
